src/realm/decimal128.cpp

```diff
@@ -1576,7 +1577,7 @@ bool Decimal128::operator>=(const Decimal128& rhs) const
     return compare(rhs) >= 0;
 }
 
-Decimal128 do_multiply(BID_UINT128 x, BID_UINT128 mul)
+static Decimal128 do_multiply(BID_UINT128 x, BID_UINT128 mul)
```

Comment:
why do we need to mark these as static?

Reply:
They correctly aren't declared in the header as they aren't intended to be used outside of this file, so the symbols shouldn't be externally visible.

Reply:
I meant why do we mark these as static instead of putting them in an anonymous namespace? This may be just like a weird mongoism but I've had "prefer anonymous namespace over static - static is a C concept, not a C++ concept" drilled into me over the years.

Reply:
Oh, right. I have mixed feelings on the static vs anonymous namespace thing. Classes have to be in anonymous namespaces and I think they're better when you have a whole group of functions, but when it's just a helper function right next to some member functions I kinda feel that an anonymous namespace is just more verbose for no benefit.

This is definitely not something I feel strongly about and I'd be fine with changing all the static functions to anonymous namepsaces if you'd prefer that.

Reply:
I'd normally go and ask folks to use an anonymous namespace as a nit comment because I tend towards lowering the surface area of C++ we use. I'd also probably nit about hoisting free functions to the top of the file so the layout of the file is includes -> anonymous globals/constants/classes/functions -> impl of whatever header we're in or main, but I don't feel strongly enough about it to hold up merging this fix.